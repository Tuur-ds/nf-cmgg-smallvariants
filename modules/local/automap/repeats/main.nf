process AUTOMAP_REPEATS {
    tag "$meta.id"
    label 'process_single'

    container "cmgg/automap:1.0.0"

    input:
    tuple val(meta), val(genome)

    output:
    tuple val(meta), path("*.bed")  , emit: repeats
    path  "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    def VERSION = "1.0.0"

    // Files are present in the container
    if (genome == "hg38" || genome == "GRCh38") {
        """
        zcat /usr/local/lib/automap/Resources/repeats_hg38.part*.bed.gz > ${prefix}.bed

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            automap: $VERSION
        END_VERSIONS
        """
    } else if (genome == "hg19" || genome == "GRCh37") {
        """
        zcat /usr/local/lib/automap/Resources/repeats.part*.bed.gz > ${prefix}.bed

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            automap: $VERSION
        END_VERSIONS
        """
    } else {
        error("No repeat regions can be found for genome '${genome}'. Available genomes are hg38/GRCh38 or hg19/GRCh37")
    }


    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    def VERSION = "1.0.0"
    if (["hg38", "GRCh38", "hg19", "GRCh37"].contains(genome)) {
        """
        touch ${prefix}.bed

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            automap: $VERSION
        END_VERSIONS
        """
    } else {
        error("No repeat regions can be found for genome '${genome}'. Available genomes are hg38/GRCh38 or hg19/GRCh37")
    }
}
