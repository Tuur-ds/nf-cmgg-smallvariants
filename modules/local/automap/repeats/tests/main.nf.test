nextflow_process {

    name "Test Process AUTOMAP_REPEATS"
    script "../main.nf"
    process "AUTOMAP_REPEATS"

    tag "modules"
    tag "modules_local"
    tag "automap"
    tag "automap/repeats"

    test("homo_sapiens - hg38") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "hg38"
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - hg38 - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "hg38"
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}