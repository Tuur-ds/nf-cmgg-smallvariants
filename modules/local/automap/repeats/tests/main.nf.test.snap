{
    "homo_sapiens - hg38": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,74190e74851226329342a3fcae133cdb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b5fc7b6bfce18165907357d8966f09d4"
                ],
                "repeats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,74190e74851226329342a3fcae133cdb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b5fc7b6bfce18165907357d8966f09d4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T16:31:33.522485317"
    },
    "homo_sapiens - hg38 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b5fc7b6bfce18165907357d8966f09d4"
                ],
                "repeats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b5fc7b6bfce18165907357d8966f09d4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T16:31:43.895708133"
    }
}